src/rpc/tensorpipe/tp_communicator.h

```diff
@@ -0,0 +1,186 @@
+/*!
```

Comment:
do we have any unit tests for these files? If have some, I think it may help new fresh like me to understand how these classes are used.